### PR TITLE
feat: Support test enviroment

### PR DESCRIPTION
**Pull request by tinsaeDev**
- [ ] All tests pass
- [ ] I have run `npm run doc`

### CODE REVIEW

**Comment by tinsaeDev** [comment]
This is a new feature, 
I have added a new feature to support testing environment.

**Comment by kamikazechaser** [comment]
change `testing` (ambiguous) to `testEnvironment`. Also remove the `yarn.lock` we generally ignore the package locks for now.

File: doc/api.md [hunk 1]
```diff
@@ -1677,7 +1678,7 @@ Animated stickers can be added to animated sticker sets and only to them:
 | --- | --- | --- | --- |
 | userId | <code>Number</code> |  | User identifier of sticker set owner |
 | name | <code>String</code> |  | Sticker set name |
-| sticker | <code>String</code> \| <code>stream.Stream</code> \| <code>Buffer</code> |  | Png image with the sticker (must be up to 512 kilobytes in size, dimensions must not exceed 512px, and either width or height must be exactly 512px), [TGS animation](https://core.telegram.org/stickers#animated-sticker-requirements) with the sticker or [WEBM video](https://core.telegram.org/stickers#video-sticker-requirements) with the sticker. |
+| sticker | <code>String</code> \| <code>stream.Stream</code> \| <code>Buffer</code> |  | Png image with the sticker (must be up to 512 kilobytes in size, dimensions must not exceed 512px, and either width or height must be exactly 512px, [TGS animation](https://core.telegram.org/stickers#animated-sticker-requirements) with the sticker or [WEBM video](https://core.telegram.org/stickers#video-sticker-requirements) with the sticker. |
```
Comment (kamikazechaser) [inline doc/api.md:1681]: Closing bracket disappeared here.

File: src/telegram.js [hunk 2]
```diff
@@ -192,7 +194,7 @@ class TelegramBot extends EventEmitter {
    * @see https://core.telegram.org/bots/api#making-requests
    */
   _buildURL(_path) {
-    return `${this.options.baseApiUrl}/bot${this.token}/${_path}`;
+    return `${this.options.baseApiUrl}/bot${this.token}${this.options.testing?"/test":""}/${_path}`;
```
Comment (kamikazechaser) [inline src/telegram.js:197]: Add some spacing in the ternary operator. We also single quotes throughout the library.

**Comment by tinsaeDev** [comment]
I've updated the source per the feedback.

**Comment by kamikazechaser** [comment]
I have pushed some fixes, could you pull my changes on your master and run `npm run doc`, then commit, push those changes. It should only add a single change that is `options.testEnvironment` to the docs.

**Review by kamikazechaser — APPROVED** [review]

**Comment by bkg866** [comment]
Hello

I hope this message finds you well. I apologize for reaching out to you directly, but I’m encountering an issue with logging in using a test number while attempting to submit my unofficial Telegram app version to the Play Store.

While I am able to log in using my personal number to the test environment, I’m having trouble with test numbers such as “+19996601234” (it works on iOS but not on Android). Please see the attached screenshot of my login attempt.

In the BuildVars file, I have set both DEBUG_VERSION and DEBUG_PRIVATE_VERSION to “true”.

Thank you for your help and assistance with this issue. I truly appreciate it.

Best regards,

**Pull request merged**